microprofile/WFLY_13129_Microprofile_CLI_Script_Example.adoc

```diff
@@ -0,0 +1,95 @@
+= Example CLI script to evolve standalone configurations with microprofile
+:author:            JF Denise
+:email:             jdenise@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+We are proposing here to define a new script in docs/examples/enable-microprofile.cli
+The script is an “example” script to be applied on default standalone configurations. 
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-13129
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* TBD
+
+=== Testing By
+
+[ ] Engineering
+
+[ ] QE
+
+=== Affected Projects or Components
+
+* WildFly
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+
+* The script is packaged in the galleon-pack maven module. It is a default package installed when WildFly is installed.
+* Installation path is _<JBOSS_HOME>/docs/examples/enable-microprofile.cli_.
+* To launch the script: _bin/jboss-cli.sh --file=docs/examples/enable-microprofile.cli_
+* By default it applies to standalone.xml, use _-Dconfig=<standalone-full.xml|standalone-ha.xml|standalone-full-ha.xml>_
+to apply to other standalone configurations.
+* The script informs the user with high level information on the modifications it applies to the configuration it is updating.
+* The script is expected to adapt to the differences that exist between various standalone configurations.
+* The script detects that a microprofile subsystem is already present in the config and skips it. 
+* The script secures the configuration using elytron security domain ApplicationDomain, http-authentication-factory and sasl-authentication-factory for items present in default configurations:
+** core-service/management
+*** access/identity security domain
+*** managementItf/http-upgrade sasl factory
+*** managementItf/ http auth factory
+** remoting/http-connector sasl factory
+** undertow 
+*** other application-domain
+*** http-invoker UNTOUCHED, references legacy security realm, no http-auth factory to set.
+** ejb3 other application-domain
+** jberet security domain
+** messaging-activemq server security-domain(full only)
+* The script remove the security subsystem
+* The script remove the management ManagementRealm. NB: the script doesn’t touch the ApplicationRealm (used by http-invoker).
+* Add Microprofile subsystems and extensions
```

Comment:
For individual extension adds it may be worth checking if they are already defined individually.

Reply:
@darranl , for each subsystem, I was thinking to check if the subsystem is present. if not present add both extension and subsystem.